src/main/java/org/jointheleague/modules/RoleBot.java

```diff
@@ -0,0 +1,143 @@
+package org.jointheleague.modules;
+
+import org.javacord.api.event.message.MessageCreateEvent;
+import org.javacord.api.listener.server.role.RoleDeleteListener;
+
+import java.awt.Color;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import org.javacord.api.entity.channel.Channel;
+import org.javacord.api.entity.channel.TextChannel;
+import org.javacord.api.entity.permission.Role;
+import org.javacord.api.entity.permission.RoleBuilder;
+import org.javacord.api.entity.server.Server;
+import org.javacord.api.entity.user.User;
+
+public class RoleBot extends CustomMessageCreateListener {
+
+	private static final String ROLE_PREFIX = "color_";
+	private final Map<String ,Role> roles;
+	private final Server server;
+	RoleBuilder rB;
+	
+	
+	
+	public RoleBot(String channelName, Optional<Server> server) {
+		super(channelName);
+		this.server=server.get();
+		rB = new RoleBuilder(this.server);
+		roles = new HashMap<>();
+		refreshRoles();
+	}
+	
+	public void refreshRoles() {
+		for(Role r : server.getRoles()) {
+			if(r.getName().startsWith(ROLE_PREFIX)) {
+				roles.put(r.getName().replaceAll(ROLE_PREFIX, ""), r);
+			}
+		}
+	}
+		
+	public void handle(MessageCreateEvent event) {
+		TextChannel channel = event.getChannel();
+		String input = event.getMessageContent();
+		User user = event.getMessageAuthor().asUser().get();
+		Server server = event.getServer().get();
```

Comment:
Do we need to get the server again here? You are already storing it as a variable on the class itself, I think we can just use that variable.

Reply:
Removed redundant server variable